### PR TITLE
[testharness] add link_to_spec and link_name properties to test().

### PR DESCRIPTION
**Pull request by tobie**
This allows idlharness to link to relevant sections of WebIDL.

### CODE REVIEW

**Comment by tobie** [comment]
Note: I voluntarily left this undocumented because of link-rot issues.

**Comment by jgraham** [comment]
Rebase this on master to pass travis.

File: resources/testharness.js [hunk 1]
```diff
@@ -2412,12 +2412,16 @@ policies and contribution forms [3].
             "<th>Message</th></tr></thead>" +
             "<tbody>";
         for (var i = 0; i < tests.length; i++) {
+            var link_name = tests[i].properties.link_name || "spec";
```
Comment (jgraham) [inline resources/testharness.js]: I think this should be documented.

**Review by jgraham — CHANGES_REQUESTED** [review]

File: resources/testharness.js (hunk 1 above)
Comment (tobie) [inline resources/testharness.js]: In `/docs/` or in the src code itself?

Comment (jgraham) [inline resources/testharness.js]: In `/docs/`

**Review by zcorpan — COMMENTED** [review]
I think some tests use a "help" property for this purpose (though not actually used by anything). Does this replace that? Should this support multiple links per test?

File: resources/testharness.js [hunk 2]
```diff
@@ -2412,12 +2412,16 @@ policies and contribution forms [3].
             "<th>Message</th></tr></thead>" +
             "<tbody>";
         for (var i = 0; i < tests.length; i++) {
+            var link_name = "see " + (tests[i].properties.link_name || "spec");
+            var link_url = tests[i].properties.link_to_spec;
+            var link_to_spec = link_url ?
+                " (<a href=" + JSON.stringify(link_url) + ">" + escape_html(link_name) + "</a>)" : "";
```
Comment (Ms2ger) [inline resources/testharness.js]: Why JSON?

**Review by sideshowbarker — COMMENTED** [review]
(Didn’t review this; was pinged by wpt-prbot to review it, so just commenting to remove it from my review queue)

**Comment by sideshowbarker** [comment]
This seems to be waiting on responses to review comments. (And will need rebasing and conflict-fixing if we are to merge it.)

**Comment by Ms2ger** [comment]
I think a proposal like this would require an RFC if it were made now, and I doubt CI will cope with this PR even if rebased, so I'll just close it. Feel free to reopen the discussion if you still believe this is useful.

**Comment by sideshowbarker** [comment]
> Feel free to reopen the discussion if you still believe this is useful.

I’m fine with having it closed — just wanted to get it resolved one way or the other

**Pull request closed without merging**